### PR TITLE
Whitelabel Icons

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: src/app/integrations/bamboo-hr/configuration/configuration.component.html [hunk 1]
```diff
@@ -48,7 +48,7 @@
                                 <ng-template pTemplate="filter" let-options="options">
                                     <div *ngIf="emails.length" class="p-inputgroup">
                                         <div class="p-input-icon-left">
-                                            <i class="pi pi-search"></i>
+                                            <app-svg-icon [svgSource]="'magnifier'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/bamboo-hr/configuration/configuration.component.html:51]: replace all magnifier with `search` and download icon, also send width and height

File: src/app/shared/components/core/svg-icon/svg-icon.component.ts [hunk 2]
```diff
@@ -47,7 +47,8 @@ const ICON_MAPPING = {
   'tabs': 'grv-budle',
   'user-plus': 'grv-person',
   'user-two': 'grv-persons',
-  'warning-outline': 'grv-caution-triangle-small'
+  'warning-outline': 'grv-caution-triangle-small',
+  'magnifier': 'grv-magnifier'
```
Comment (ashwin1111) [inline src/app/shared/components/core/svg-icon/svg-icon.component.ts:51]: search-medium: grv-magnifier-medium

File: src/app/shared/components/core/svg-icon/svg-icon.component.ts [hunk 3]
```diff
@@ -47,7 +47,8 @@ const ICON_MAPPING = {
   'tabs': 'grv-budle',
   'user-plus': 'grv-person',
   'user-two': 'grv-persons',
-  'warning-outline': 'grv-caution-triangle-small'
+  'warning-outline': 'grv-caution-triangle-small',
```
Comment (ashwin1111) [inline src/app/shared/components/core/svg-icon/svg-icon.component.ts:50]: arrow-left is missing

File: src/assets/icons/arrow-left.svg
Comment (ashwin1111) [inline src/assets/icons/arrow-left.svg:1]: 16*16

File: src/app/shared/components/helper/paginator/paginator.component.html [hunk 4]
```diff
@@ -29,40 +29,40 @@
...
                 <div class="">
                     <button type="button" class="tw-h-32-px tw-w-32-px prevBtn tw-flex tw-place-items-center tw-pt-2-px tw-mr-12-px" (click)="navigateToPage(1)" [disabled]="page === 1" >
-                        <i class="pi pi-minus tw-rotate-90"></i><i class="pi pi-chevron-left"></i>
+                        <i class="pi pi-minus tw-rotate-90"></i><app-svg-icon [svgSource]="'arrow-tail-left'"></app-svg-icon>
                     </button>
                 </div>
                 <div class="">
                     <button type="button" class="tw-h-32-px tw-w-32-px prevBtn tw-mr-8-px" (click)="previousPageChange(page)" [disabled]="page === 1" >
-                        <i class="pi pi-chevron-left tw-w-6-px tw-h-12-px tw-pt-6-px"></i>
+                        <app-svg-icon [svgSource]="'arrow-tail-left'" [styleClasses]="'tw-w-6-px tw-h-12-px tw-pt-6-px'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/paginator/paginator.component.html:52]: arrow-left

File: src/app/shared/components/helper/paginator/paginator.component.html [hunk 5]
```diff
@@ -29,40 +29,40 @@
...
-                        <i class="pi pi-angle-right tw-w-6-px tw-h-12-px tw-pt-6-px"></i>
+                        <app-svg-icon [svgSource]="'arrow-right'" [styleClasses]="'tw-w-6-px tw-h-12-px tw-pt-6-px'"></app-svg-icon>
                     </button>
                 </div>
             </div>
             <div *ngIf="pageType !== PaginatorPage.DASHBOARD" class="tw-flex tw-justify-between">
                 <div class="">
                     <button type="button" class="tw-h-32-px tw-w-32-px prevBtn tw-flex tw-place-items-center tw-pt-2-px tw-mr-12-px" (click)="navigateToPage(1)" [disabled]="page === 1" >
-                        <i class="pi pi-minus tw-rotate-90"></i><i class="pi pi-chevron-left"></i>
+                        <i class="pi pi-minus tw-rotate-90"></i><app-svg-icon [svgSource]="'arrow-tail-left'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/paginator/paginator.component.html:47]: arrow-bar-left

File: src/app/shared/components/helper/paginator/paginator.component.html [hunk 6]
```diff
@@ -29,40 +29,40 @@
...
                     <button type="button" class="tw-h-32-px tw-w-30-px nextBtn" (click)="nextPageChange(page)" [disabled]="page === totalPages">
-                        <i class="pi pi-angle-right tw-w-6-px tw-h-12-px tw-pt-6-px"></i>
+                        <app-svg-icon [svgSource]="'arrow-right'" [styleClasses]="'tw-w-6-px tw-h-12-px tw-pt-6-px'"></app-svg-icon>
                     </button>
                 </div>
             </div>
             <div *ngIf="pageType !== PaginatorPage.DASHBOARD" class="tw-flex tw-justify-between">
                 <div class="">
                     <button type="button" class="tw-h-32-px tw-w-32-px prevBtn tw-flex tw-place-items-center tw-pt-2-px tw-mr-12-px" (click)="navigateToPage(1)" [disabled]="page === 1" >
-                        <i class="pi pi-minus tw-rotate-90"></i><i class="pi pi-chevron-left"></i>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/paginator/paginator.component.html:47]: search with `pi pi-` in whole repo and update them as well

File: src/app/shared/components/helper/paginator/paginator.component.html [hunk 7]
```diff
@@ -29,40 +29,40 @@
...
+                        <app-svg-icon [svgSource]="'arrow-tail-left'" [styleClasses]="'tw-w-6-px tw-h-12-px tw-pt-6-px'"></app-svg-icon>
                     </button>
                 </div>
                 <div class="tw-w-fix tw-h-32-px tw-flex tw-items-center tw-justify-center tw-border-2-px tw-border-separator tw-py-8-px tw-px-12-px tw-rounded-4-px tw-bg-white tw-mr-10-px">
                     {{page}} of {{totalPages}}
                 </div>
                 <div >
                     <button type="button" class="tw-h-32-px tw-w-32-px nextBtn tw-mr-12-px" (click)="nextPageChange(page)" [disabled]="page === totalPages">
-                        <i class="pi pi-chevron-right tw-w-6-px tw-h-12-px tw-pt-6-px"></i>
+                        <app-svg-icon [svgSource]="'arrow-tail-right'" [styleClasses]="'tw-w-6-px tw-h-12-px tw-pt-6-px'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/paginator/paginator.component.html:60]: fix name

File: src/app/shared/components/helper/paginator/paginator.component.html [hunk 8]
```diff
@@ -29,40 +29,40 @@
...
                     </button>
                 </div>
                 <div>
                     <button type="button" class="tw-h-32-px tw-w-32-px tw-ml-10-px nextBtn tw-flex tw-place-items-center tw-pt-2-px" (click)="navigateToPage(totalPages)" [disabled]="page === totalPages">
-                        <i class="pi pi-chevron-right"></i><i class="pi pi-minus tw-rotate-90"></i>
+                        <app-svg-icon [svgSource]="'arrow-tail-left'"></app-svg-icon><i class="pi pi-minus tw-rotate-90"></i>
```
Comment (ashwin1111) [inline src/app/shared/components/helper/paginator/paginator.component.html:65]: fix name

File: src/app/integrations/bamboo-hr/configuration/configuration.component.html [hunk 9]
```diff
@@ -48,7 +48,7 @@
                                 <ng-template pTemplate="filter" let-options="options">
                                     <div *ngIf="emails.length" class="p-inputgroup">
                                         <div class="p-input-icon-left">
-                                            <i class="pi pi-search"></i>
+                                            <app-svg-icon [svgSource]="'search'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/bamboo-hr/configuration/configuration.component.html:51]: width and height not added

File: src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html [hunk 10]
```diff
@@ -133,7 +133,7 @@ <h5 *ngIf="(formControllerName === 'reimbursableExportType' || formControllerNam
                             <span>
                                 {{ item.value }}
                             </span>
-                            <i class="pi pi-times" (click)="removeFilter(form.controls[formControllerName])"></i>
+                            <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(form.controls[formControllerName])"></i>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html:136]: let's use svg component

File: src/app/shared/components/configuration/email-multi-select-field/email-multi-select-field.component.html [hunk 11]
```diff
@@ -31,7 +31,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
             <ng-template pTemplate="filter" let-options="options">
                 <div *ngIf="emails.length" class="p-inputgroup">
                     <div class="p-input-icon-left p-input-icon-right">
-                        <i class="pi pi-search !tw-text-placeholder"></i>
+                        <app-svg-icon [svgSource]="brandingConfig.brandId === 'fyle' ? 'search' : 'grv-search'" [styleClasses]="'!tw-text-placeholder'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/email-multi-select-field/email-multi-select-field.component.html:34]: still in old fashion, need not send dynamic source, send only fyle source

File: src/app/shared/components/configuration/email-multi-select-field/email-multi-select-field.component.html [hunk 12]
```diff
@@ -73,7 +73,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
         <div class="tw-p-24-px tw-border-b-1-px tw-border-b-box-color">
             <div class="tw-flex tw-justify-between tw-items-center">
                 <p class="tw-text-slightly-normal-text-color tw-text-20-px tw-font-500">Add new Email Address</p>
-                <i class="pi pi-times tw-cursor-pointer" style="font-size: 12px" (click)="closeModel()"></i>
+                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" class="tw-cursor-pointer" style="font-size: 12px" (click)="closeModel()"></i>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/email-multi-select-field/email-multi-select-field.component.html:76]: use svg component

File: src/app/shared/components/core/svg-icon/svg-icon.component.ts [hunk 13]
```diff
@@ -47,7 +47,12 @@ const ICON_MAPPING = {
...
+  'search-medium': 'grv-search-medium',
+  'arrow-left': 'grv-chevron-left',
+  'arrow-right': 'grv-chevron-right',
+  'arrow-bar-left': 'grv-chevron-first',
+  'arrow-bar-right': 'grv-chevron-last'
```
Comment (ashwin1111) [inline src/app/shared/components/core/svg-icon/svg-icon.component.ts:55]: are all these 24*24? if not consider naming it -medium

File: src/app/shared/components/export-log/export-log-filter/export-log-filter.component.html [hunk 14]
```diff
@@ -1,7 +1,7 @@
 <div [formGroup]="exportLogForm" class="tw-flex tw-items-center tw-p-24-px">
 
     <span class="custom-search-field" [ngStyle]="{ 'border-color': isSearchFocused ? 'black' : '#ccc' }">
-        <i class="pi pi-search"></i>
+        <app-svg-icon [svgSource]="'search'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/export-log/export-log-filter/export-log-filter.component.html:4]: width height is missing

File: src/assets/icons/co/grv-search-medium.svg
Comment (ashwin1111) [inline src/assets/icons/co/grv-search-medium.svg:1]: name says -medium, but size is 24*24

Comment (anishfyle) [reply src/assets/icons/co/grv-search-medium.svg:1]: will make it 16*16

File: src/app/integrations/bamboo-hr/configuration/configuration.component.html [hunk 15]
```diff
@@ -48,7 +48,7 @@
                                 <ng-template pTemplate="filter" let-options="options">
                                     <div *ngIf="emails.length" class="p-inputgroup">
                                         <div class="p-input-icon-left">
-                                            <i class="pi pi-search"></i>
+                                            <app-svg-icon [svgSource]="'search'" [height]="'16px'" [width]="'16px'"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/bamboo-hr/configuration/configuration.component.html:51]: I see only search-medium in svg component, why passing search is working fine?

File: src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html [hunk 16]
```diff
@@ -161,7 +161,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.value }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(advancedSettingsForm.controls.defaultLocation)"></i>
+                                <app-svg-icon [svgSource]="'cross-medium'" (click)="removeFilter(advancedSettingsForm.controls.defaultLocation)"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html:164]: height width is missing

File: src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html [hunk 17]
```diff
@@ -182,7 +182,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.value }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(advancedSettingsForm.controls.defaultDepartment)"></i>
+                                <app-svg-icon [svgSource]="'cross-medium'" (click)="removeFilter(advancedSettingsForm.controls.defaultDepartment)"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html:185]: height width is missing

File: src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html [hunk 18]
```diff
@@ -203,7 +203,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.value }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(advancedSettingsForm.controls.defaultProject)"></i>
+                                <app-svg-icon [svgSource]="'cross-medium'" (click)="removeFilter(advancedSettingsForm.controls.defaultProject)"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html:206]: height width is missing

File: src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html [hunk 19]
```diff
@@ -220,7 +220,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.value }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(advancedSettingsForm.controls.defaultClass)"></i>
+                                <app-svg-icon [svgSource]="'cross-medium'" (click)="removeFilter(advancedSettingsForm.controls.defaultClass)"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html:223]: height width is missing

File: src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html [hunk 20]
```diff
@@ -237,7 +237,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.value }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(advancedSettingsForm.controls.defaultItems)"></i>
+                                <app-svg-icon [svgSource]="'cross-medium'" (click)="removeFilter(advancedSettingsForm.controls.defaultItems)"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/integrations/intacct/intacct-shared/intacct-advanced-settings/intacct-advanced-settings.component.html:240]: height width is missing

File: src/app/shared/components/configuration/configuration-schedule-export/configuration-schedule-export.component.html [hunk 21]
```diff
@@ -37,7 +37,7 @@ <h5 class="!tw-text-faded-text-color tw-text-14-px tw-pt-8-px !tw-font-400 !tw-l
                                 <span>
                                     {{ item.label }}
                                 </span>
-                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-chevron-down fyle' : 'pi pi-chevron-down co']"></i>
+                                <i [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-chevron-down fyle' : 'pi pi-chevron-down co']"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-schedule-export/configuration-schedule-export.component.html:40]: use svg component

File: src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html [hunk 22]
```diff
@@ -50,7 +50,7 @@ <h5 *ngIf="(formControllerName === 'reimbursableExportType' || formControllerNam
                             <span>
                                 {{ item.label }}
                             </span>
-                            <i *ngIf="showClearIcon && !isDisabled" [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(form.controls[formControllerName])"></i>
+                            <i *ngIf="showClearIcon && !isDisabled" [ngClass]="[brandingConfig.brandId === 'fyle' ? 'pi pi-times fyle' : 'pi pi-times co']" (click)="removeFilter(form.controls[formControllerName])"></app-svg-icon>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-select-field/configuration-select-field.component.html:53]: opens with <i but ends with svg

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**